src/Maestro/DependencyUpdateErrorProcessor/DependencyUpdateErrorProcessor.cs

```diff
@@ -0,0 +1,386 @@
+                    else
+                    {
+                        // Check if the error for the method ProcessPendingUpdatesAsync already exists 
+                        var argumentParser = ParseIssueCommentBody(body);
+                        if (string.Equals(argumentParser["method"], "ProcessPendingUpdatesAsync") &&
+                            string.Equals(argumentParser["errorMessage"], updateHistoryError.ErrorMessage))
+                        {
+                            return new Tuple<string, bool>(description.Append(genericDescription).ToString(), true);
+                        }
+                    }
+                    // creates a new description body since the issue does not exists previously. 
+                    return new Tuple<string, bool>(description.Append(genericDescription).ToString(), false);
+                }
+                default:
+                    return new Tuple<string, bool>(description.Append(genericDescription).ToString(), false);
+            }
+        }
+
+        /*      Eg: Body of Comment :
+                The following errors have been detected when attempting to update dependencies in 'https://github.com/maestro-auth-test/maestro-test2' 
+
+                [marker]: <> (subscriptionId: 'a9f0536e-15cc-4d1f-79a4-08d79acf8bba', method: 'UpdateAssetsAsync', errorMessage: 'build: 14	Unexpected error processing action: Validation Failed')
+                **SubscriptionId:** 'a9f0536e-15cc-4d1f-79a4-08d79acf8bba'
+                **Source Repository :**  'https://github.com/maestro-auth-test/maestro-test1'
+                **Target Repository :**  'https://github.com/maestro-auth-test/maestro-test2'
+                **Branch Name :**  '38'
+                **Error Message :**  'build: 14	Unexpected error processing action: Validation Failed'
+                **Method :** 'UpdateAssetsAsync'
+                **Action :** 'Updating assets for subscription: ee8cdcfb-ee51-4bf3-55d3-08d79538f94d'
+                **Last seen :** 'Maestro.Data.RepositoryBranchUpdateHistoryEntry'
+
+                The Regex : (\w+): '([^,\n]+)'(?:, |\)) captures (subscriptionId: 'a9f0536e-15cc-4d1f-79a4-08d79acf8bba', method: 'UpdateAssetsAsync', errorMessage: 'build: 14	Unexpected error processing action: Validation Failed')
+                the above string and puts it into two groups. 
+
+                group[1]            group[2]
+                subscriptionId      a9f0536e-15cc-4d1f-79a4-08d79acf8bba
+                method              UpdateAssetsAsync
+                errorMessage        build: 14	Unexpected error processing action: Validation Failed
+
+                This is put into a dictionary, where group[1] is put in as key and group[2] as value.
+        */
+        private static Dictionary<string, string> ParseIssueCommentBody (string body)
+        {
+            var arguments = new Dictionary<string, string>();
+            var matches = CommentMarker.Matches(body);
```

Comment:
suggest using type here.

Reply:
still not resolved.

Reply:
changed the way I capture the parameters